### DISCUSSION TITLE
default compressibility factors

### DISCUSSION THREAD

**Discussion by jkisse** [e2nIEE/pandapipes#304] [General]
I would like to discuss the default-compressibility factors that are stored in properties/[fluid name]/compressibility.txt in the format _slope, offset._ 
For H and L natural gas, the slope (=der_comp) is currently -0.0022, i.e. Z=0.78 @ 100bar. According to [this figure on ResearchGate](https://www.researchgate.net/figure/Compressibility-Factor-of-Natural-Gas_fig2_317840680) this means that we are on the 0° C isothermal curve. At 25 °C, Z would be 0.85, according to the chart.

We could also choose the 10 °C curve (-0.0019) as default since it seems closer to real operational conditions. In the long term, we could also implement different factors for a range of temperatures. What do you think?

This is in particular relevant for the calculation of high pressure networks, as the error is proportional to the pressure.

**Comment by jkisse** [comment]
This could be solved by nested properties (der_compr = f(p,T))

**Comment by jkisse** [comment]
We could also consider a generic description of Z, for example by applying the CNGA-method that calculates Z from pressure, temperature and relative density. An example is [[Liu et al., 2020](https://doi.org/10.1002/aic.16908)]
![grafik](https://user-images.githubusercontent.com/44288328/90258387-c3ab0c80-de48-11ea-9caa-610f61f8ec43.png)

**Comment by dlohmeier** [comment]
We will compare our results to other tools and if we find that it makes sense, we change the approach, otherwise we stay with der_comp=-0.022 which is at 0°C.

**Comment by jkisse** [comment]
According to [Clees et al., 2021](https://doi.org/10.1016/j.enconman.2021.113984), the GERG-2008 EoS [[Kunz & Wagner, 2012](https://pubs.acs.org/doi/10.1021/je300655b)] yields best results for mixtures and pure hydrogen. An implementation would be way more complex than the linear approach.
